conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/DropRuleInterpreter.scala

```diff
@@ -15,21 +15,25 @@
 
 package za.co.absa.enceladus.conformance.interpreter.rules
 
-import za.co.absa.enceladus.model.conformanceRule.DropConformanceRule
-import org.apache.spark.sql.Dataset
-import org.apache.spark.sql.Row
-import org.apache.spark.sql.SparkSession
-import za.co.absa.enceladus.dao.EnceladusDAO
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
 import za.co.absa.enceladus.conformance.CmdConfig
-import za.co.absa.enceladus.utils.transformations.{ArrayTransformations, DeepArrayTransformations}
+import za.co.absa.enceladus.dao.EnceladusDAO
+import za.co.absa.enceladus.model.conformanceRule.DropConformanceRule
+import za.co.absa.enceladus.utils.schema.SchemaUtils
+import za.co.absa.enceladus.utils.transformations.DeepArrayTransformations
 
 case class DropRuleInterpreter(rule: DropConformanceRule) extends RuleInterpreter {
 
   def conform(df: Dataset[Row])(implicit spark: SparkSession, dao: EnceladusDAO, progArgs: CmdConfig): Dataset[Row] = {
-    if (rule.outputColumn.contains('.')) {
-      conformNestedField(df)
+    if (SchemaUtils.fieldExists(rule.outputColumn, df.schema)) {
+      if (rule.outputColumn.contains('.')) {
+        conformNestedField(df)
+      } else {
+        conformRootField(df)
+      }
     } else {
-      conformRootField(df)
+      log.warn(s"Could not drop ${rule.outputColumn}. Column is not present in the dataset")
```

Comment:
Sorry for the late comment. Didn't notice when looked at it the first time. Are you ok with the warning here? Should it be a validation exception? True, that if a column does not exists we might ignore the 'drop'.  But shouldn't we disallow it anyway? No strong opinion on this, just a suggestion.

Found `validateFieldExistence(...)` method used for similar checks for other conformance rules. Of course, there the existence of an input field is critical.

Reply:
I have thought about it and came to a conclusion to follow spark api for now. Spark api does not throw anything in case of dropping column that does not exist. 

I am more than happy to change it if needed/wanted.